src/domain/challenge/challenge/pages/InnovationFlow/ChallengeInnovationFlowView.tsx

```diff
@@ -1,20 +1,55 @@
 import { Grid } from '@mui/material';
-import React, { FC } from 'react';
-import { useUrlParams } from '../../../../../hooks';
-import { useChallengeProfileInfoQuery } from '../../../../../hooks/generated/graphql';
+import React, { FC, useMemo } from 'react';
+import { useApolloErrorHandler, useUrlParams } from '../../../../../hooks';
+import {
+  refetchChallengeLifecycleQuery,
+  useChallengeProfileInfoQuery,
+  useHubLifecycleTemplatesQuery,
+  useUpdateChallengeInnovationFlowMutation,
+} from '../../../../../hooks/generated/graphql';
 import Loading from '../../../../../common/components/core/Loading/Loading';
-import EditLifecycle from '../../../../platform/admin/templates/InnovationTemplates/EditLifecycle';
+import UpdateInnovationFlow from '../../../../platform/admin/templates/InnovationTemplates/UpdateInnovationFlow';
 import ChallengeLifecycleContainer from '../../../../../containers/challenge/ChallengeLifecycleContainer';
+import { LifecycleType } from '../../../../../models/graphql-schema';
+import { SelectInnovationFlowFormValuesType } from '../../../../platform/admin/templates/InnovationTemplates/SelectInnovationFlowDialog';
 
 const ChallengeInnovationFlowView: FC = () => {
   const { challengeNameId = '', hubNameId = '' } = useUrlParams();
+  const handleError = useApolloErrorHandler();
+
+  const { data: hubLifecycleTemplates } = useHubLifecycleTemplatesQuery({
+    variables: { hubId: hubNameId },
+  });
+  const innovationFlowTemplates = hubLifecycleTemplates?.hub?.templates?.lifecycleTemplates;
+  const filteredInnovationFlowTemplates = innovationFlowTemplates?.filter(
+    template => template.type === LifecycleType.Challenge
+  );
 
   const { data: challengeProfile } = useChallengeProfileInfoQuery({
     variables: { hubId: hubNameId, challengeId: challengeNameId },
     skip: false,
   });
   const challenge = challengeProfile?.hub?.challenge;
-  const challengeId = challenge?.id || '';
+  const challengeId = useMemo(() => challenge?.id || '', [challenge]);
```

Comment:
Generally speaking, this is an over-optimization - It should be used on "heavy functions".
I would recommend removing this and using it as in the below suggestions.